wasm-text.md

```diff
@@ -360,6 +360,229 @@ Imports can be declared like regular functions, memories, etc., by giving an inl
+```
+
+#### KWasm Administrative Instructions
+
+The following instructions are not part of the official Wasm text format.
+They are currently supported in KWasm text files, but may be deprecated.
+
+```k
+    rule #t2aInstr<C>(trap) => trap
+
+    rule #t2aInstr<C>(block VT:VecType IS:Instrs end) => block VT #t2aInstrs<C>(IS) end
+
+    rule #t2aInstr<_>(init_local I V) => init_local I V
+    rule #t2aInstr<_>(init_locals VS) => init_locals VS
+```
+
+### List Functions
+
+The following are helper functions.
+They distribute the text-to-abstract functions above over lists.
+
+```k
+    syntax Stmts      ::= "#t2aStmts"      "<" Context ">" "(" Stmts      ")" [function]
+    syntax Defns      ::= "#t2aDefns"      "<" Context ">" "(" Defns      ")" [function]
+    syntax Instrs     ::= "#t2aInstrs"     "<" Context ">" "(" Instrs     ")" [function]
+    syntax LocalDecls ::= "#t2aLocalDecls" "<" Context ">" "(" LocalDecls ")" [function]
+ // ------------------------------------------------------------------------------------
+    rule #t2aStmts<C>(S:Stmt SS:Stmts) => #t2aStmt<C>(S) #t2aStmts<C>(SS)
+    rule #t2aStmts<_>(.Stmts) => .Stmts
+
+    rule #t2aDefns<C>(D:Defn DS:Defns) => #t2aDefn<C>(D) #t2aDefns<C>(DS)
+    rule #t2aDefns<_>(.Defns) => .Defns
+
+    rule #t2aInstrs<C>(I:Instr IS:Instrs) => #t2aInstr<C>(I) #t2aInstrs<C>(IS)
+    rule #t2aInstrs<_>(.Instrs) => .Instrs
+
+    rule #t2aLocalDecls<C>(LD:LocalDecl LDS:LocalDecls) => #t2aLocalDecl<C>(LD) #t2aLocalDecls<C>(LDS)
+    rule #t2aLocalDecls<_>(.LocalDecls) => .LocalDecls
+```
+
+### Functions for Gathering Context
+
+The following are helper functions for gathering and updating context.
+
+```k
+    syntax Map        ::= #ids2Idxs(TypeUse, LocalDecls)      [function, functional]
+                        | #ids2Idxs(Int, TypeUse, LocalDecls) [function, functional]
+ // --------------------------------------------------------------------------------
+    rule #ids2Idxs(TU, LDS) => #ids2Idxs(0, TU, LDS)
+
+    rule #ids2Idxs(_, .TypeDecls, .LocalDecls) => .Map
+    rule #ids2Idxs(N, (type _)    , LDS) => #ids2Idxs(N, .TypeDecls, LDS)
+    rule #ids2Idxs(N, (type _) TDS, LDS) => #ids2Idxs(N, TDS       , LDS)
+
+    rule #ids2Idxs(N, (param ID:Identifier _) TDS, LDS)
+      => (ID |-> N) #ids2Idxs(N +Int 1, TDS, LDS)
+    rule #ids2Idxs(N, (param _) TDS, LDS)   => #ids2Idxs(N +Int 1, TDS, LDS)
+    rule #ids2Idxs(N, TD:TypeDecl TDS, LDS) => #ids2Idxs(N       , TDS, LDS) [owise]
+
+    rule #ids2Idxs(N, .TypeDecls, local ID:Identifier _ LDS:LocalDecls)
+      => (ID |-> N) #ids2Idxs(N +Int 1, .TypeDecls, LDS)
+    rule #ids2Idxs(N, .TypeDecls, LD:LocalDecl LDS) => #ids2Idxs(N +Int 1, .TypeDecls, LDS) [owise]
+
+    syntax FuncSpec   ::= "#updating" "(" FuncSpec ")" | "#ready" "(" FuncSpec ")"
+ // ------------------------------------------------------------------------------
```

Comment:
I don't think these should be needed, and either way they should be moved closer to where they are used (and the formatting fixed).